src/OVAL/probes/unix/linux/systemdunitproperty.c

```diff
@@ -251,6 +259,44 @@ static int unit_callback(const char *unit, void *cbarg)
 	return 0;
 }
 
+void get_unit_file_state_in_offline_mode(const char *unit, struct unit_callback_vars *vars)
+{
+	char systemctl_call[] = "systemctl list-unit-files | grep -P '\\S+\\.service'";
```

Comment:
This ***cannot** be here. We can't call external programs!

Reply:
Is replacing this by calls to the DBUS a viable alternative?

Reply:
I think we  have to implement something filesystem based.

We already do something like that eg. in process probe, where we read form '/proc/' directory to get attributes of processes.

You can try to have a look how systemd implements `systemctl list-unit-files`.

From a quick look I found this:
https://github.com/systemd/systemd/blob/fbd0b64f44a7bae678137d67ad5da202f6b8d809/src/systemctl/systemctl.c

However I don't know if their code can be reused, that needs to be investigated.